pipeline/ephys.py

```diff
@@ -0,0 +1,282 @@
+        depth:       decimal(6, 2) # (um) manipulator depth relative to surface of the brain (0); more ventral is more negative
+        theta:       decimal(5, 2) # (deg) - elevation - rotation about the ml-axis [0, 180] - w.r.t the z+ axis
+        phi:         decimal(5, 2) # (deg) - azimuth - rotation about the dv-axis [0, 360] - w.r.t the x+ axis
+        beta:        decimal(5, 2) # (deg) rotation about the shank of the probe [-180, 180] - clockwise is increasing in degree - 0 is the probe-front facing anterior
+        """
+
+    class RecordableBrainRegion(dj.Part):
+        definition = """
+        -> master
+        -> lab.BrainArea
+        -> lab.Hemisphere
+        """
+
+    class InsertionNote(dj.Part):
+        definition = """
+        -> master
+        ---
+        insertion_note: varchar(1000)
+        """
+
+    class RecordingSystemSetup(dj.Part):
+        definition = """
+        -> master
+        ---
+        sampling_rate: int  # (Hz)
+        """
+
+
+@schema
+class LFP(dj.Imported):
+    definition = """
+    -> ProbeInsertion
+    ---
+    lfp_sample_rate: float          # (Hz)
+    lfp_time_stamps: longblob       # timestamps with respect to the start of the recording (recording_timestamp)
+    lfp_mean: longblob              # mean of LFP across electrodes
+    """
+
+    class Channel(dj.Part):
+        definition = """  
+        -> master
+        -> lab.ElectrodeConfig.Electrode
+        ---
+        lfp: longblob           # recorded lfp at this electrode
+        """
+
+
+@schema
+class UnitQualityType(dj.Lookup):
+    definition = """
+    # Quality
+    unit_quality  :  varchar(100)
+    ---
+    unit_quality_description :  varchar(4000)
+    """
+    contents = [
+        ('good', 'single unit'),
+        ('ok', 'probably a single unit, but could be contaminated'),
+        ('multi', 'multi unit'),
+        ('all', 'all units')
+    ]
+
+
+@schema
+class CellType(dj.Lookup):
+    definition = """
+    #
+    cell_type  :  varchar(100)
+    ---
+    cell_type_description :  varchar(4000)
+    """
+    contents = [
+        ('Pyr', 'putative pyramidal'),
+        ('FS', 'fast spiking'),
+        ('not classified', 'intermediate spike-width that falls between spike-width thresholds for FS or Putative pyramidal cells'),
+        ('all', 'all types')
+    ]
+
+
+@schema
+class ClusteringMethod(dj.Lookup):
+    definition = """
+    clustering_method: varchar(16)
+    """
+
+    contents = zip(['jrclust_v3', 'kilosort', 'jrclust_v4', 'kilosort2'])
+
+
+@schema
+class Clustering(dj.Imported):
+    definition = """
+    -> ProbeInsertion
+    -> ClusteringMethod
+    ---
+    clustering_time: datetime  # time of generation of this set of clustering results 
+    quality_control: bool  # has this clustering results undergone quality control
+    manual_curation: bool  # is manual curation performed on this clustering result
```

Comment:
Currently I'm using KS2 for clustering and JRClust for manual curation. How should we record this?